src/tiled/mapdocument.h

```diff
@@ -315,6 +328,8 @@ private slots:
     MapRenderer *mRenderer;
     Layer* mCurrentLayer;
     MapObjectModel *mMapObjectModel;
+    TerrainModel *mTerrainModel;
```

Comment:
You've removed the getter, but this member should not be added either, nor should the forward declaration and the `tilesetTerrain...` set of signals.

Since your `WangSetModel` now follows the same logic as the `TerrainModel`, it should also not be part of the `MapDocument`, nor should the `tilesetWangSet...` set of signals.